homework-3/src/main/java/com/epam/cashier/controller/service/repository/impl/ProductRepositoryImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.epam.cashier.controller.service.repository.impl;
+
+import com.epam.cashier.controller.service.model.Product;
+import com.epam.cashier.controller.service.repository.ProductRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class ProductRepositoryImpl implements ProductRepository {
+    private final List<Product> productList = new ArrayList<>();
+
+    @Override
+    public List<Product> getAllProducts() {
+        return new ArrayList<>(productList);
+    }
+
+    @Override
+    public Product getProduct(int id) {
+        return productList
+                .stream()
+                .filter(product -> product.getProductId() == id)
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("product is not found!"));
```

Comment:
Please create custom Exception for this and other cases, for example ProductNotFoundException

Reply:
Fixed.